webapp/store/views.py

```diff
@@ -132,26 +133,17 @@ def search_snap():
         if not snap_searched and not snap_category:
             return flask.redirect(flask.url_for(".homepage"))
 
-        size = flask.request.args.get("limit", default=24, type=int)
-        offset = flask.request.args.get("offset", default=0, type=int)
+        # The default size should be 48 (rows of 4)
+        size = 48
 
-        try:
-            page = floor(offset / size) + 1
-        except ZeroDivisionError:
-            size = 10
-            page = floor(offset / size) + 1
+        # Page 1 has a snap at the top, and a few rows of 3, followed by rows
```

Comment:
It seems that that is the case only when showing first category page.
First page of standard search (like: https://snapcraft-io-canonical-web-and-design-pr-1890.run.demo.haus/search?q=snap) should not be limited to 47 snaps.

If the page size varies between pages - are we sure we are not skipping some results when calculating what to get from API?

Reply:
Aaah good catch with normal search results! Forgot about those :D!

Total results is 141.
Page 1 has 47, so the last page is calculated as 3 exactly.
Page 2 has 48, so the last page is calculated as 2.9375, which is rounded up to 3
Winning

Total results is 144.
Page 1 has 47, so the last page is calculated as 3.063...., which is rounded up to 4
Page 2 has 48, so the last page is calculated as 3 exactly.
Uh-oh

I'm of the opinion that the number of people that skip to the last page is minimal and the chance of this happening is low. If someone does jump beyond the "last" page, the snaps from the last page are shown - so nobody will see a blank screen (https://snapcraft-io-canonical-web-and-design-pr-1890.run.demo.haus/search?q=snap&page=5010202010)